tests/middleware/auth/auth_store_test.py

```diff
@@ -0,0 +1,74 @@
+import pytest
+from sqlalchemy.exc import IntegrityError, NoResultFound # type: ignore
+
+from src.data_access.models.user_model import User
+from src.middleware.auth.auth_store import AuthStore, IncorrectCredentialsError
+from src.middleware.exceptions.auth.user_already_exists_error import \
+    UserAlreadyExistsError
+from src.middleware.exceptions.auth.user_not_found_error import \
+    UserNotFoundError
+
+
+@pytest.fixture
+def mock_user(mocker):
+    mock_user = mocker.Mock()
+    mock_user.one.return_value = mock_user
```

Comment:
I dont know if naming is indicative..
u can write these two lines in one line - mock_user.one.return_value.filter.return_value = ...